Source/JavaScriptCore/runtime/StringPrototypeInlines.h

```diff
@@ -89,12 +90,10 @@ ALWAYS_INLINE JSString* stringSlice(JSGlobalObject* globalObject, VM& vm, JSStri
     }
 }
 
-ALWAYS_INLINE std::tuple<int32_t, int32_t> extractSubstringOffsets(int32_t length, int32_t startValue, std::optional<int32_t> endValue)
+ALWAYS_INLINE std::pair<int32_t, int32_t> extractSubstringOffsets(int32_t length, int32_t startValue, std::optional<int32_t> endValue)
 {
-    int32_t start = std::min<int32_t>(std::max<int32_t>(startValue, 0), length);
-    int32_t end = length;
-    if (endValue)
-        end = std::min<int32_t>(std::max<int32_t>(endValue.value(), 0), length);
+    int32_t start = std::clamp<int32_t>(startValue, 0, length);
+    int32_t end = std::clamp<int32_t>(endValue.value_or(length), 0, length);
```

Comment:
I wish we could always use `std::clamp` instead of `std::min` combined with `std::max`. But it’s not _always_ OK, because if the lower bound is higher than the upper bound the behavior of `std::clamp` is undefined (I was shocked to discover this), and in most cases it returns the upper bound in this case, whereas `std::min` and `std::max` can be used in a way that guarantees the result is the lower bounds in that case, which is typically the more useful behavior in my experience.

But in this case, since `length` is guaranteed to be >= 0, it’s fine to use `std::clamp`.

I suggest we not write out the types explicitly, because everything here is `int32_t`, including the 0, since in practice int is always `int32_t` in all the configurations we support.

```suggestion
    int32_t start = std::clamp(startValue, 0, length);
    int32_t end = std::clamp(endValue.value_or(length), 0, length);
```

Reply:
Updated